modules/deprecations.nix

```
@@ -0,0 +1,40 @@
+{lib, ...}: let
+  inherit (builtins) match trace;
+  inherit (lib.asserts) assertMsg;
+  inherit (lib.modules) doRename;
+  inherit (lib.options) showOption;
+  inherit (lib.lists) concatLists;
+
+  mkRenamedOptionModuleUntil = from: to: date: let
+    isValidISODate = dateString: (match "[0-9]{4}-[0-9]{2}-[0-9]{2}" dateString != null);
+  in
+    assert assertMsg (isValidISODate date) "Cannot rename `${showOption from}` to `${showOption to}`: Your date needs to be in ISO 8601 format, ie. yyyy-mm-dd (got ${date} instead).";
+      doRename {
+        inherit from to;
+        visible = false;
+        warn = true;
+        use = trace "Obsolete option `${showOption from}' is used. It was renamed to `${showOption to}', and will be removed on ${date}";
+      };
+in {
+  imports = concatLists [
+    [
+      # removed options
+    ]
+    [
+      # deprecations
+      (mkRenamedOptionModuleUntil
+        ["rum" "programs" "zsh" "integrations" "starship" "enable"]
+        ["rum" "programs" "starship" "integrations" "zsh" "enable"]
+        "2025-09-20")
+      (mkRenamedOptionModuleUntil
+        ["rum" "programs" "hyprland"]
+        ["rum" "desktops" "hyprland"]
+        "2025-09-20")
+
+      (mkRenamedOptionModuleUntil
+        ["rum" "gtk"]
+        ["rum" "misc" "gtk"]
+        "2025-09-20")
```

Nitpick, but I feel like this makes it a bit more readable.

```suggestion
      (mkRenamedOptionModuleUntil
        ["rum" "programs" "zsh" "integrations" "starship" "enable"]
        ["rum" "programs" "starship" "integrations" "zsh" "enable"]
        "2025-09-20"
      )
      (mkRenamedOptionModuleUntil
        ["rum" "programs" "hyprland"]
        ["rum" "desktops" "hyprland"]
        "2025-09-20"
      )
      (mkRenamedOptionModuleUntil
        ["rum" "gtk"]
        ["rum" "misc" "gtk"]
        "2025-09-20"
      )
```

---

Not possible with alenajdra. It's either ```nix
(
foo
)

or: 
```nix
(foo)
```